lib/semian.rb

```diff
@@ -154,7 +157,12 @@ def resources
 require 'semian/protected_resource'
 require 'semian/unprotected_resource'
 require 'semian/platform'
-if Semian.sysv_semaphores_supported? && Semian.semaphores_enabled?
+require 'semian/shared_memory_object'
+require 'semian/sliding_window'
+require 'semian/atomic_integer'
+require 'semian/atomic_enum'
+Semian.extension_loaded = Semian.sysv_semaphores_supported? && Semian.semaphores_enabled?
+if Semian.extension_loaded
```

Comment:
This basically means we now have three different ways to check for SysV: `Semian.sysv_semaphores_supported?`, `Semian.semaphores_enabled?` and now `Semian.extension_loaded`. IMO calling `Semian.semaphores_enabled?` should be enough here, but it will have to check that they're supported:

```ruby
  def semaphores_enabled?
    !ENV['SEMIAN_SEMAPHORES_DISABLED'] && Semian.sysv_semaphore_supported?
  end
```